src/module_ncio.f90

```diff
@@ -1,81 +1,58 @@
 !> @file
-!! Module for reading/writing netcdf files such as the lat/lon grid history files output by the GFS.
-!! writing requires a template file.
+!! @brief Module for reading/writing netcdf files such as the lat/lon grid history files output by the GFS..
+!! @author jeff whitaker <jeffrey.s.whitaker@noaa.gov> @date 201910
+
+!> writing requires a template file.
 !! Handles 32 and 64 bit real variables, 8, 16 and 32 bit integer
 !! variables and char variables. Variables can have up to 5 dimensions.
 !! @author jeff whitaker <jeffrey.s.whitaker@noaa.gov> @date 201910
 module module_ncio
-!
+
   use netcdf
   use mpi
-!
+  
   implicit none
   private
-!
+
   type Variable
-! netCDF variable ID
-     integer varid 
-! number of dimensions
-     integer ndims 
-! netCDF data type
-     integer dtype 
-! number of attributes
-     integer natts 
-! compression level (if > 0)
-     integer deflate_level 
-! shuffle filter?
-     logical shuffle  
-! has an unlimited dim?
-     logical hasunlim 
-! variable name
-     character(len=nf90_max_name) :: name 
-! netCDF dimension IDs
-     integer, allocatable, dimension(:) :: dimids 
-!
-! indices into Dataset%dimensions for associated dimensions.
-     integer, allocatable, dimension(:) :: dimindxs
-! names of associated dimensions.
-     character(len=nf90_max_name), allocatable, dimension(:) :: dimnames
-! current dimension lengths (updated after every write_vardata call)
+     integer varid !< NetCDF variable ID.
+     integer ndims !< Number of dimensions.
+     integer dtype !< netCDF data type
+     integer natts !< number of attributes
+     integer deflate_level !< compression level (if > 0)
+     logical shuffle  !< shuffle filter?
+     logical hasunlim !< has an unlimited dim?
+     character(len=nf90_max_name) :: name !< variable name
+     integer, allocatable, dimension(:) :: dimids !< netCDF dimension IDs
+
+     integer, allocatable, dimension(:) :: dimindxs !< indices into Dataset%dimensions for associated dimensions.
+     character(len=nf90_max_name), allocatable, dimension(:) :: dimnames !< names of associated dimensions.
+     !< current dimension lengths (updated after every write_vardata call)
```

Comment:
This needs to be on the next line. The !< comment marker means "apply this doxygen comment to the preceding code object."